src/python/pants/backend/python/lint/pylint/rules_integration_test.py

```diff
@@ -0,0 +1,151 @@
+from pants.engine.fs import FileContent, InputFilesContent, Snapshot
+from pants.engine.legacy.structs import PythonTargetAdaptor
+from pants.engine.rules import RootRule
+from pants.engine.selectors import Params
+from pants.rules.core.lint import LintResult
+from pants.source.wrapped_globs import EagerFilesetWithSpec
+from pants.testutil.option.util import create_options_bootstrapper
+from pants.testutil.test_base import TestBase
+
+
+class PylintIntegrationTest(TestBase):
+  source_root = "tests/python/pants_test"
+  good_source = FileContent(path="good.py", content=b"'''docstring'''\nVAR = 42\n")
+  bad_source = FileContent(path="bad.py", content=b"VAR = 42\n")
+
+  create_python_library = partialmethod(
+    TestBase.create_library, path=source_root, target_type="python_library", name="target",
+  )
+
+  def write_file(self, file_content: FileContent) -> None:
+    self.create_file(
+      relpath=str(PurePath(self.source_root, file_content.path)),
+      contents=file_content.content.decode(),
+    )
+
+  @classmethod
+  def alias_groups(cls) -> BuildFileAliases:
+    return BuildFileAliases(targets={'python_library': PythonLibrary})
+
+  @classmethod
+  def rules(cls):
+    return (*super().rules(), *pylint_rules(), RootRule(PylintTarget))
+
+  def run_pylint(
+    self,
+    source_files: List[FileContent],
+    *,
+    config: Optional[str] = None,
+    passthrough_args: Optional[str] = None,
+    interpreter_constraints: Optional[str] = None,
+    skip: bool = False,
+  ) -> LintResult:
+    args = ["--backend-packages2=pants.backend.python.lint.pylint"]
+    if config:
+      # TODO: figure out how to get this file to exist...
+      self.create_file(relpath="pylintrc", contents=config)
+      args.append("--pylint-config=pylintrc")
+    if passthrough_args:
+      args.append(f"--pylint-args='{passthrough_args}'")
+    if skip:
+      args.append(f"--pylint-skip")
+    input_snapshot = self.request_single_product(Snapshot, InputFilesContent(source_files))
+    target = PylintTarget(
+      PythonTargetAdaptor(
+        sources=EagerFilesetWithSpec(self.source_root, {'globs': []}, snapshot=input_snapshot),
+        address=Address.parse(f"{self.source_root}:target"),
+        compatibility=[interpreter_constraints] if interpreter_constraints else None,
+      )
+    )
+    return self.request_single_product(
+      LintResult, Params(target, create_options_bootstrapper(args=args)),
+    )
+
+  def test_single_passing_source(self) -> None:
+    self.create_python_library()
+    self.write_file(self.good_source)
+    result = self.run_pylint([self.good_source])
+    assert result.exit_code == 0
+    assert "Your code has been rated at 10.00/10" in result.stdout.strip()
+
+  def test_single_failing_source(self) -> None:
+    self.create_python_library()
+    self.write_file(self.bad_source)
+    result = self.run_pylint([self.bad_source])
+    assert result.exit_code == 16
```

Comment:
Huh, TIL that Pylint uses 16 for failures..

Reply:
Could this be documented in a comment? How would I find this out?

Reply:
Good point. I added comments for the nonzero error codes and a link at the top of the class.